src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -193,17 +178,39 @@ public object GetOrCreateObjectForComInstance(IntPtr externalComObject, CreateOb
             => (comWrappersImpl ?? s_globalInstance!).CreateObject(externalComObject, flags);
 
         /// <summary>
-        /// Called when a request is made for a collection of objects to be released.
+        /// Get the currently registered managed object or uses the supplied managed object and registers it.
         /// </summary>
-        /// <param name="objects">Collection of objects to release.</param>
+        /// <param name="externalComObject">Object to import for usage into the .NET runtime.</param>
+        /// <param name="flags">Flags used to describe the external object.</param>
+        /// <param name="wrapper">The <see cref="object"/> to be used as the wrapper for the external object</param>
+        /// <returns>Returns a managed object associated with the supplied external COM object.</returns>
         /// <remarks>
-        /// The default implementation of this function throws <see cref="System.NotImplementedException"/>.
+        /// If the <paramref name="wrapper"/> instance already has an associated external object a <see cref="System.NotSupportedException"/> will be thrown.
         /// </remarks>
-        protected virtual void ReleaseObjects(IEnumerable objects)
+        public object GetOrRegisterObjectForComInstance(IntPtr externalComObject, CreateObjectFlags flags, object wrapper)
```

Comment:
Should this check that the wrapper is non-null?

Reply:
Boo. Forgot that the enforcement is compiler not functional. Yes it should.